app/components/primer/alpha/dialog.rb

```diff
@@ -0,0 +1,142 @@
+# frozen_string_literal: true
+
+module Primer
+  module Alpha
+    # A `Dialog` is used to remove the user from the main application flow,
+    # to confirm actions, ask for disambiguation or to present small forms.
+    #
+    # @accessibility
+    #   - **Dialog Accessible Name**: A dialog should have an accessible name,
+    #   so screen readers are aware of the purpose of the dialog when it opens.
+    #   Give an accessible name setting `:title`. The accessible name will be
+    #   used as the main heading inside the dialog.
+    #   - **Dialog unique id**: A dialog should be unique. Give a unique id
+    #   setting `:dialog_id`. If no `:dialog_id` is given, a default randomize
+    #   hex id is generated.
+    #
+    #   The combination of both `:title` and `:dialog_id` establishes an
+    #   `aria-labelledby` relationship between the title and the unique id of
+    #   the dialog.
+    class Dialog < Primer::Component
+      DEFAULT_WIDTH = :medium
+      WIDTH_MAPPINGS = {
+        :small => "Overlay--width-small",
+        DEFAULT_WIDTH => "Overlay--width-medium",
+        :large => "Overlay--width-large",
+        :xlarge => "Overlay--width-xlarge",
+        :xxlarge => "Overlay--width-xxlarge"
+      }.freeze
+      WIDTH_OPTIONS = WIDTH_MAPPINGS.keys
+
+      DEFAULT_HEIGHT = :auto
+      HEIGHT_MAPPINGS = {
+        :small => "Overlay--height-small",
+        DEFAULT_HEIGHT => "Overlay--height-auto",
+        :large => "Overlay--height-large",
+        :xlarge => "Overlay--height-xlarge",
+      }.freeze
+      HEIGHT_OPTIONS = HEIGHT_MAPPINGS.keys
+
+      # Optional button to open the dialog.
+      #
+      # @param system_arguments [Hash] The same arguments as <%= link_to_component(Primer::ButtonComponent) %>.
+      renders_one :show_button, lambda { |**system_arguments|
+        system_arguments[:classes] = class_names(
+          system_arguments[:classes]
+        )
+        system_arguments[:id] = "dialog-show-#{@system_arguments[:id]}"
+        system_arguments["data-show-dialog-id"] = @system_arguments[:id]
+        system_arguments[:data] = (system_arguments[:data] || {}).merge({ "show-dialog-id": @system_arguments[:id] })
+        Primer::ButtonComponent.new(**system_arguments)
+      }
+
+      # Header content.
+      #
+      # @param hide_divider [Boolean] If true the visual dividing line between the header and body will be hidden
+      # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+      renders_one :header, lambda { |hide_divider: false, **system_arguments|
+        Primer::Alpha::Dialog::Header.new(
+          id: @id,
+          title: @title,
+          subtitle: @subtitle,
+          hide_divider: hide_divider,
+          **system_arguments
+        )
+      }
+
+      # Required body content.
```

Comment:
Is this a required section any more given the introduction of the `content` in the `.erb` file?

Reply:
It is as the content is used for wrapping outside of the body content.